stable/database/files/nuosm

```diff
@@ -387,6 +387,69 @@ function resolveLatestSource() {
   return 0
 }
 
+#=======================================
+# function - Looks for a previous archive and journal and imports it
+#
+function loadFromSnapshot() {
+  local RECREATE_ARCHIVES="false"
+
+  if [ -z "$(ls -A ${DB_DIR})" ]; then
```

Comment:
This is just checking that the archive directory is not empty?

I think we should the same thing we do in `nuocmd start sm`, which is look for `$DB_DIR/1.atm`, which contains the master catalog. See the Python snippet below from `nuodocker.py`:

```python
    def _has_archive_data(self, directory_path):
        return os.path.exists(os.path.join(directory_path, "1.atm"))
```

@madscientist, is it still safe with LSA to look for `1.atm`?

Reply:
I prefer checking for `1.cat` instead of `info.json` because it help in a very weird race condition: if the SM is being restored from a snapshot, the archive is moved, and the `info.json` is deleted, but nuosm crashes before it can register the archive with the admin and write a new `info.json`

If we look for `info.json`, the SM will be left in a state where it is trying to find an archive to copy, cannot find another one in the volume, and crashes.

If we look for `1.cat`, the SM will start properly, though there is a small risk that the archive ends up registered twice with the admin.

Honestly, it is a 1 in a million chance that this specific timing will ever actually happen, but I am one to take defensive programming a bit too far.

Reply:
@sivanov-nuodb Non-LSA archives don't always have an `1.cat`. Talking to Paul S, he suggested looking for `state.dat`.

Reply:
Agreed. We should also update the code snippet I posted. I created a JIRA for the `nuodocker start sm` change.

Reply:
@agoelzer recently changed [nuosm](https://github.com/nuodb/nuodb-helm-charts/blob/master/stable/database/files/nuosm#L497) to check for `1.cat`. It seems that this needs to be changed as well.